### ISSUE TITLE
hook function doesn't work sometimes 

### ISSUE THREAD

**Issue by Lorenzo978** [pawn-lang/YSI-Includes#522]
It's hard to explain, but I have a function that I hook on multiple files using `hook function Player_HasVehicleKey` and sometimes, after i re-compile for other modifies that doesn't involve directly this function, like adding other `hook function` or `hook`,  all the hook chain doesn't get called. I show you in pratice:

This is my function:
```pawn
stock Player_HasVehicleKey(playerid, vehicleid)
{
    #pragma unused playerid
    if(Iter_Contains(Iter_Vehicles, vehicleid) && vehicleInfo[vehicleid][vType] == VEHICLE_TYPE_GENERIC)
        return true;
    
    return false;
}
```

This is one of the `hook function` I re-call:
```pawn
hook function Player_HasVehicleKey(playerid, vehicleid)
{
    if(Vehicle_GetType(vehicleid) == VEHICLE_TYPE_FACTION)
        return Player_GetFaction(playerid) && Player_GetFaction(playerid) == Vehicle_GetFactionID(vehicleid);
    
    return continue(playerid, vehicleid);
}
```

Now, let's say I add this code:
```pawn
hook function SetVehiclePos(vehicleid, Float:x, Float:y, Float:z)
{
    if(Vehicle_IsEngineTimerOn(vehicleid))
        vehicleInfo[vehicleid][vCalculateDistance] = true;

    return continue(vehicleid, x,y,z);
}
```
If I recompile, it's **probably** (it doesn't happen always) that all the `hook function Player_HasVehicleKey(playerid, vehicleid)` aren't called and the only code execute is:
```pawn
  #pragma unused playerid
  if(Iter_Contains(Iter_Vehicles, vehicleid) && vehicleInfo[vehicleid][vType] == VEHICLE_TYPE_GENERIC)
      return true;
  
  return false;
```

It's sounds weird, I know it, but I can't figure it out what is going on.

**Comment by Y-Less** [comment]
"probably" isn't much use, but inconsistent compiling sounds like an old compiler bug - are you on the latest one?  Anyway, without a reproducible MCVE to test there's not a lot anyone can do.

**Comment by Lorenzo978** [comment]
> "probably" isn't much use, but inconsistent compiling sounds like an old compiler bug - are you on the latest one?

Yes, I'm using the 3.10.10 version. Could be something related to sampctl?

> Anyway, without a reproducible MCVE to test there's not a lot anyone can do.

I know, but it's difficult because is something interconnected with all the modular project.
